setup-env.sh

```diff
@@ -8,9 +8,17 @@ else
     machine="$( "$parent_dir"/src/machine.sh )"
 fi
 
-export SPACK_SYSTEM_CONFIG_PATH="$parent_dir"/sysconfigs/"$machine"
-export SPACK_USER_CONFIG_PATH="$parent_dir"/user-config
-export SPACK_USER_CACHE_PATH="$parent_dir"/user-cache
+if [[ ${machine:0:1} == "/" ]]; then
+    export SPACK_SYSTEM_CONFIG_PATH="$machine"/config
+    export SPACK_USER_CONFIG_PATH="$parent_dir"/sysconfigs/uenv
+    export SPACK_USER_CACHE_PATH="$parent_dir"/user-cache/uenv
+else
+    export SPACK_SYSTEM_CONFIG_PATH="$parent_dir"/sysconfigs/"$machine"
+    export SPACK_USER_CONFIG_PATH="$parent_dir"/user-config
+    export SPACK_USER_CACHE_PATH="$parent_dir"/user-cache/"$machine"
+fi
```

Comment:
This introduces unnecessary complexity on `SPACK_USER_CACHE_PATH`. `export SPACK_USER_CACHE_PATH="$parent_dir"/user-cache` is sufficient because the spec of a package carries all dependencies, including OS, architecture, etc. A separation by machine name or environment is therefore not necessary.

Reply:
Now use a single cache again.

Reply:
It is still a very strange error I get.
No dependency from the uenv is used in a non uenv build, and vice-versa.

Also it used to work for a long time, just at some point we got this persistent error.

But maybe this will be different on a machine where we only use uenv, bo fixed mounted sw-stack.